### PR TITLE
Some miscellaneous Python3 fixes.

### PR DESCRIPTION
**Pull request by prabhuramachandran**
The changes should be compatible with Python 2.

### CODE REVIEW

**Comment by corranwebster** [comment]
Thanks for these: the code should be being run through 2to3 on install if Python 3 is being used, which should automatically fix all of these, but it would be really nice one day to not have to rely on 2to3.

**Comment by prabhuramachandran** [comment]
Thanks, the collections change is not seen by 2to3.  There might be a few more small changes needed and I will push those too.

File: pyface/ui/qt4/workbench/split_tab_widget.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 # Standard library imports.
 import sys
 
+if sys.version_info[0] > 2:
+    unicode = str
```
Comment (corranwebster) [inline pyface/ui/qt4/workbench/split_tab_widget.py]: `str` here is being used to coerce the output of `QSplitter.saveState()` which is a `QByteArray`.  From fixing up the TraitsUI code, I found that using `QByteArray`'s `data()` method worked well across versions to give an appropriate Python object (`str` in 2.x, `bytes` in 3.x).

Comment (corranwebster) [inline pyface/ui/qt4/workbench/split_tab_widget.py]: Offline discussion determined that I was misunderstanding here.

**Comment by corranwebster** [comment]
Going to merge this now - we can always open another one later.

**Pull request merged**